src/Umbraco.Web.UI.Client/src/packages/core/entity-action/common/create/create.action.ts

```diff
@@ -5,42 +5,55 @@ import { UMB_ENTITY_CREATE_OPTION_ACTION_LIST_MODAL } from './modal/constants.js
 import { UMB_MODAL_MANAGER_CONTEXT } from '@umbraco-cms/backoffice/modal';
 import type { UmbControllerHost } from '@umbraco-cms/backoffice/controller-api';
 import { umbExtensionsRegistry } from '@umbraco-cms/backoffice/extension-registry';
-import { createExtensionApi, UmbExtensionsManifestInitializer } from '@umbraco-cms/backoffice/extension-api';
-import type { ManifestEntityCreateOptionAction } from '@umbraco-cms/backoffice/entity-create-option-action';
+import {
+	type UmbExtensionManifestInitializer,
+	createExtensionApi,
+	UmbExtensionsManifestInitializer,
+	type PermittedControllerType,
+} from '@umbraco-cms/backoffice/extension-api';
+import type {
+	ManifestEntityCreateOptionAction,
+	UmbEntityCreateOptionAction,
+} from '@umbraco-cms/backoffice/entity-create-option-action';
 
 export class UmbCreateEntityAction extends UmbEntityActionBase<MetaEntityActionCreateKind> {
 	#hasSingleOption = true;
-	#singleActionOptionManifest?: ManifestEntityCreateOptionAction;
+	#promise?: Promise<void>;
+	#singleOptionApi?: UmbEntityCreateOptionAction;
 
 	constructor(host: UmbControllerHost, args: UmbEntityActionArgs<MetaEntityActionCreateKind>) {
 		super(host, args);
 
-		new UmbExtensionsManifestInitializer(
-			this,
-			umbExtensionsRegistry,
-			'entityCreateOptionAction',
-			(ext) => ext.forEntityTypes.includes(this.args.entityType),
-			async (actionOptions) => {
-				this.#hasSingleOption = actionOptions.length === 1;
-				this.#singleActionOptionManifest = this.#hasSingleOption
-					? (actionOptions[0].manifest as unknown as ManifestEntityCreateOptionAction)
-					: undefined;
-			},
-			'umbEntityActionsObserver',
-		);
+		this.#promise = new Promise((resolve) => {
```

Comment:
I think this could be replaced with as `.asPromise()`

Reply:
Or is this because all Extension should be resolved before you like the promise to be complete.. Maybe this is a feature we need for the Property Value Presets as well.. so maybe add a TODO above so we get back to this one.

Reply:
I have added a comment. There are two parts to this. The main issue is that there is an async method in the callback. We need to wait for the result of that before we are "happy" 😄

Reply:
Makes sense, thanks